src/c++/perf_analyzer/inference_profiler.cc

```diff
@@ -816,7 +827,8 @@ InferenceProfiler::ProfileHelper(
 }
 
 bool
-InferenceProfiler::DetermineStability(LoadStatus& load_status)
+InferenceProfiler::DetermineStability(
+    LoadStatus& load_status, bool check_latency)
```

Comment:
When is check_latency ever false?

Reply:
We decided not to require latency to stabilize for the async case given the variability of latency in that case. If latency does not stabilize (but throughput does) by the end of the runs, then we want to print an error but still allow PA to stabilize.

See this line for more information: https://github.com/triton-inference-server/client/pull/688/files#diff-1d869f26de26a6908d10260f21bd433d63ab260cd58156aed650f563001547d1R796

Reply:
i missed DetermineStability passing in false on that line. Thank you.
Would lines 796 - 802 technically be considered determining stability? Should that be internalized to the method and remove check_latency as an arg, atleast to the DetermineStability method?

Reply:
It's still determining stability, just only of throughput. Are you recommending creating another DetermineStability function that only checks throughput?

Reply:
Note that we still call DetermineStability for async models. If it doesn't ever stabilize, then we check that throughput stabilizes... if so, we allow it while printing a warning to the user.

Reply:
I was thinking this logic around that space felt like it was all part of DeterminingStability.
I think we have 2 levels of checks now.
-first order: determine stability (called each pass?)
-second order: if not stable but in async mode, try again ignoring latency.

if i understand it correctly, what you are doing seems correct. Just feels like its all part of determiningStability.

This is probably highlighting a need for a refactor at this point, which we knew for PA already.
I think we can close this comment as you clearly thought this through. Just feels like every addition makes this code base harder to follow.